src/api/app/controllers/webui/users/notifications_controller.rb

```diff
@@ -91,9 +82,9 @@ def paginated_notifications
   end
 
   def notifications_filter
-    NotificationsFilterPresenter.new(projects_for_filter,
+    NotificationsFilterPresenter.new(ProjectFilterChoicesPresenter.new(params[:project]),
+                                     GroupFilterChoicesPresenter.new,
```

Comment:
What do you think of this kind of refactor? The thing is, with the group filtering now there are 5 parameters. Rubocop complains about that. I've been thinking in disabling that cop here but first I want to try this approach.

Reply:
Better if we could switch from positional argument to keywords. In another hand if you are depending exclusively on params and coupling the classes together isn't a problem, then you could pass params to the `NotificationsFilterPresenter` and in the initializer you create the dependency with the `ProjectFilterChoicesPresenter` and `GroupFilterChoicesPresenter`